deepinv/optim/prior.py

```diff
@@ -0,0 +1,170 @@
+import torch
+import torch.nn as nn
+
+from deepinv.optim.utils import gradient_descent
+
+
+class Prior(nn.Module):
+    r"""
+    Prior term :math:`g{x}`. To implement a custom prior, for an explciit prior, overwrite `g` (do not forget to specify `self.explicit_prior = True`). For an implicit prior, overwrite the `grad` or `prox`.
```

Comment:
I think it should be Negative log prior term :math:`g(x)`

Reply:
is there a way to avoid setting the explicit_prior flat? e.g., if g doesn't exist just raise an error?